### DISCUSSION TITLE
Building librdkafka for Unreal Engine

### DISCUSSION THREAD

**Discussion by shinyclaw** [edenhill/librdkafka#3668] [Q&A]
Hi all, I'm trying to build librdkafka using Unreal's native Linux toolchain (https://docs.unrealengine.com/4.27/en-US/SharingAndReleasing/Linux/NativeToolchain/) for later usage in the Unreal Linux server plugin. In CMake, I'm setting the CMAKE_C_COMPILER as "v12_clang-6.0.1-centos7/x86_64-unknown-linux-gnu/bin/clang", and CMAKE_CXX_COMPILER as "v12_clang-6.0.1-centos7/x86_64-unknown-linux-gnu/bin/clang++", but, after configure the following errors appears:

    Building C object CMakeFiles/cmTC_4d8ce.dir/testCCompiler.c.o
    /home/shiny/Dokumenty/v12_clang-6.0.1-centos7/x86_64-unknown-linux-gnu/bin/clang    -MD -MT CMakeFiles/cmTC_4d8ce.dir/testCCompiler.c.o -MF CMakeFiles/cmTC_4d8ce.dir/testCCompiler.c.o.d -o CMakeFiles/cmTC_4d8ce.dir/testCCompiler.c.o -c /home/shiny/Dokumenty/librdkafka-1.6.2/out/CMakeFiles/CMakeTmp/testCCompiler.c
    Linking C executable cmTC_4d8ce
    /snap/cmake/1000/bin/cmake -E cmake_link_script CMakeFiles/cmTC_4d8ce.dir/link.txt --verbose=1
    /home/shiny/Dokumenty/v12_clang-6.0.1-centos7/x86_64-unknown-linux-gnu/bin/clang -rdynamic CMakeFiles/cmTC_4d8ce.dir/testCCompiler.c.o -o cmTC_4d8ce 
    /home/shiny/Dokumenty/v12_clang-6.0.1-centos7/x86_64-unknown-linux-gnu/bin/x86_64-unknown-linux-gnu-ld: /usr/lib/gcc/x86_64-linux-gnu/9/../../../x86_64-linux-gnu/crt1.o: unrecognized relocation (0x2a) in section `.text'
    /home/shiny/Dokumenty/v12_clang-6.0.1-centos7/x86_64-unknown-linux-gnu/bin/x86_64-unknown-linux-gnu-ld: final link failed: Bad value
    clang: error: linker command failed with exit code 1 (use -v to see invocation)

Can anybody tell me how to adjust other CMake settings to successfully generate make files?

**Comment by rolandyoung** [comment]
I think you need to find experts on CMake and/or the Unreal toolchain rather than asking here. 
It looks as if the configured CMake project is failing when building a test target to check that the C Compiler works and generates an object file that can be linked: this is a basic test of the build environment as you have configured it.
So you need to ask an Unreal toolchain expert why that target (in particular) won't build - they (and you) can ignore the rest of the project until that target builds.
